ddtrace/internal/flare/handler.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any
+from typing import Callable
+from typing import List
+
+from ddtrace.internal.flare.flare import Flare
+from ddtrace.internal.flare.flare import FlareSendRequest
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def _tracerFlarePubSub():
+    from ddtrace.internal.flare._subscribers import TracerFlareSubscriber
+    from ddtrace.internal.remoteconfig._connectors import PublisherSubscriberConnector
+    from ddtrace.internal.remoteconfig._publishers import RemoteConfigPublisher
+    from ddtrace.internal.remoteconfig._pubsub import PubSub
+
+    class _TracerFlarePubSub(PubSub):
+        __publisher_class__ = RemoteConfigPublisher
+        __subscriber_class__ = TracerFlareSubscriber
+        __shared_data__ = PublisherSubscriberConnector()
+
+        def __init__(self, callback: Callable, flare: Flare):
+            self._publisher = self.__publisher_class__(self.__shared_data__, None)
+            self._subscriber = self.__subscriber_class__(self.__shared_data__, callback, flare)
+
+    return _TracerFlarePubSub
+
+
+def _handle_tracer_flare(flare: Flare, data: dict, cleanup: bool = False):
+    if cleanup:
+        flare.revert_configs()
+        flare.clean_up_files()
+        return
+
+    if "config" not in data:
+        log.warning("Unexpected tracer flare RC payload %r", data)
+        return
+    if len(data["config"]) == 0:
+        log.warning("Unexpected number of tracer flare RC payloads %r", data)
+        return
+
+    product_type = data.get("metadata", [{}])[0].get("product_name")
```

Comment:
same here, we can probably be a little more verbose to be more clear and try to avoid the defaults of `[{}]`.

Could be the same for `configs` too, since if we don't have any configs, then we don't need to call the prepare functions.